src/pages/trpc-example.tsx

```diff
@@ -0,0 +1,79 @@
+import { NextPage } from "next";
+import { LinearProgress, Select, Typography } from "@material-ui/core";
+import { trpcClient } from "../trpc-client";
+import { PropsWithChildren, useEffect, useState } from "react";
+
+const Wrapper = ({ children }: PropsWithChildren<{}>) => (
+  <div style={{ display: "grid", gridAutoFlow: "column" }}>{children}</div>
+);
+
+const Products = ({
+  products,
+}: {
+  products: Array<{
+    node: {
+      id: string;
+      name: string;
+    };
+  }>;
```

Comment:
Can't we use generated fragment type or at least define type outside of function, like:
```ts
const Products = ({
  products,
}: ProductsProps) => ...
```
?

Reply:
Extracted type and added comment. I didn't use fragment (because its tRPC between graphQL and frontend), I linked to tRPC inference docs but didn't implement them because its a lot of boilerplate which is not important for this example